### ISSUE TITLE
Autocomplete only works intermittently using React

### ISSUE THREAD

**Issue by looeee** [tailwindlabs/tailwindcss-intellisense#392]
Autocomplete is very inconsistently applied. 
What happens is that when I start to type a class name, e.g. `h-` then I get the list: `h-0`, `h-1` etc. as expected:

![working](https://user-images.githubusercontent.com/5307958/130011579-d879f30b-de45-491c-9b59-8945906c6161.png)

However, if I do anything other than accept a suggestion (e.g. hit backspace, move the cursor to another line, switch files and back etc.) then all I get is a "Loading...." modal that never completes. 

![loading](https://user-images.githubusercontent.com/5307958/130011589-9af31daf-f695-4c42-a241-ea0c9471996d.png)

It doesn't matter how long I wait, it never completes. 
I'm currently working in WSL on Windows, however I was getting this behavior on a standard Windows project as well. 

Tested with all other extensions except Remote-WSL disabled. 
Tested with VSCode 1.59

Tested with a few versions of this extension: 
0.6.14 
0.6.0 
0.5.10
0.5.0 

The extension output panel:

```
Found Tailwind config file: /home/lewy/dev/decanter/apps/tank-map/tailwind.config.js
Found tailwindcss v2.2.4: /home/lewy/dev/decanter/node_modules/tailwindcss
Initialised successfully.
```

And my tailwind config:

``` js
const { join } = require('path');
const defaultTheme = require('tailwindcss/defaultTheme');

const brandColors = {
	primary: '#324556',
	brandBlue: '#319DD7',
	brandBlueLight: '#CAE6F4',
	brandGreenDark: '#049549',
	brandGreen: '#00AA5E',
	brandGreenLight: '#B3E3CA',
};

module.exports = {
	purge: [join(__dirname, './src/**/*.jsx'), join(__dirname, './src/**/*.tsx'), join(__dirname, './src/**/*.html')],
	darkMode: false, // or 'media' or 'class'
	theme: {
		fontFamily: {
			sans: ['roboto', ...defaultTheme.fontFamily.sans],
			serif: [...defaultTheme.fontFamily.serif],
			mono: [...defaultTheme.fontFamily.mono],
			opensans: ['Open Sans'],
		},
		backgroundColor: (theme) => ({
			...theme('colors'),
			...brandColors,
		}),
		borderColor: (theme) => ({
			...theme('colors'),
			...brandColors,
		}),
		textColor: (theme) => ({
			...theme('colors'),
			...brandColors,
		}),
		extend: {},
	},
	variants: {
		extend: {},
	},
	plugins: [],
};

```

**Comment by bradlc** [comment]
Hey @looeee. Does it work if you manually trigger completions (`ctrl+space`)? Also, in your last screenshot it looks like the `Loading...` is actually because of a hover provider, rather than a completion provider.

Typically the issue you are seeing is resolved by setting `editor.quickSuggestions.strings` to `true`, as mentioned in the extension readme:

```
"editor.quickSuggestions": {
  "strings": true
}
```

**Comment by looeee** [comment]
> Does it work if you manually trigger completions (ctrl+space)? 

Yes, that seems to work. I'd forgotten about that hotkey so this is decent workaround, thank you.  

> Typically the issue you are seeing is resolved by setting editor.quickSuggestions.strings to true, as mentioned in the extension readme:

Sorry, forgot to mention that in my post. This is my settings.json:

``` js
  "editor.quickSuggestions": {
    "other": true,
    "comments": true,
    "strings": true
  },
```

Just to make sure, I commented out all other `editor.xxx` lines except for that on in my `settings.json` and reloaded - it doesn't make a difference. 

> Also, in your last screenshot it looks like the Loading... is actually because of a hover provider, rather than a completion provider.

Yeah, I guess this is the hover preview not showing up. I wasn't aware these were different.

**Comment by AgainPsychoX** [comment]
I have the similar issue. 

![Animation](https://user-images.githubusercontent.com/23557839/130368178-a720ca74-5f76-470d-bf7f-57dae9e18231.gif)

**Comment by looeee** [comment]
@AgainPsychoX  that looks like the same issue, thanks for the video. 

Can you share some more details of your project? Are you using Typescript? 

@bradlc from examining the process log, looks like what might be happening is that this is running with too low priority, only after all the typescript checking is done. Do you think that's possible? And is there any way to increase the priority of this extension to run before type checking?

**Comment by bradlc** [comment]
> I have the similar issue.
> 
> ![Animation](https://user-images.githubusercontent.com/23557839/130368178-a720ca74-5f76-470d-bf7f-57dae9e18231.gif)

@AgainPsychoX This looks exactly like what I would expect if you didn't have `editor.quickSuggestions.strings` set to `true`. Can you confirm that you do?

@looeee is there any chance you can share a minimal repository that reproduces the issue? I am unable to reproduce with just your Tailwind config, so maybe it's to do with something else in your project like TypeScript like you suggested.

**Comment by looeee** [comment]
@bradlc I just tried it with this [next.js+typescript+tailwind starter](https://github.com/theodorusclarence/ts-nextjs-tailwind-starter) and can confirm that I get the same issue there. I'm running on Ubuntu on WSL2. 

To create a repo with the starter use this command:

```
npx create-next-app -e https://github.com/theodorusclarence/ts-nextjs-tailwind-starter project-name
```

**Comment by junglast** [comment]
@looeee 
I had the exactly same issue.
But after changing vscode settings.json like yours autocomplete works again.

```
{
  "css.validate": false,
  "editor.quickSuggestions": {
    "other": true,
    "comments": true,
    "strings": true
  }
}
```

My project is also based on next.js+typescript+tailwind starter,
and using Mac OS.

**Comment by junglast** [comment]
I have the exactly same issue.

My project is also based on next.js+typescript+tailwind starter, and using mac os & vscode 1.61.2.

**Comment by alexbu92** [comment]
Same here, Next.js 12 + typescript + tailwind

**Comment by jhirn** [comment]
So i'm not sure if an update or series of plugins changes happened. WIth the same config i'm auto completing just fine if I'm starting from a new word. For instance if I type `w-1` i get all of the completions with the hints and everything. If I then hit the backspace key, it goes away and I don't even get it back if I type 1. It's a super minor annoyance, I usual delete by word anyways. 

I'm not even 100% if it's supposed to work like that but I'd encourage anyone having issue here to make sure they're starting with a new style. Almost all of my CSS is in .tsx files.

**Comment by guiazca** [comment]
Have you updated de VSCode? I have updated and it worked

**Comment by looeee** [comment]
Just checked again - I'm on the latest VSCode (1.63.2) and this issue still remains for me.

**Comment by guiazca** [comment]
> Just checked again - I'm on the latest VSCode (1.63.2) and this issue still remains for me.

on fist className for me does not work, but trying on second its worked, like
className="h-2" don't work in this but
className="h-2 justify-center" work, after h-2

**Comment by lensbart** [comment]
Same issue here. What information can I provide to help debug this issue?

* v0.7.7
* VSCode 1.64.2
* no plugin-specific config

**Comment by lacherogwu** [comment]
Same issue here

**Comment by lensbart** [comment]
@lacherogwu your image doesn’t seem to load anymore, but it did before, and it showed the exact problematic behaviour indeed. (I think you can host images directly on GitHub by dragging-and-dropping it in the comment field.)

**Comment by lacherogwu** [comment]
Fixed

> @lacherogwu your image doesn’t seem to load anymore, but it did before, and it showed the exact problematic behaviour indeed. (You can host images directly on GitHub by dragging-and-dropping it in the comment field.)

**Comment by aryankarim** [comment]
I have the same issue! When it doesn't work, there is no force in this universe to make it work.

**Comment by mattchannn** [comment]
> I have the same issue! When it doesn't work, there is no force in this universe to make it work.

I can use Ctrl+Space and force the auto completion comes up. You can try this as a workaround...

**Comment by mattchannn** [comment]
> I have the same issue! When it doesn't work, there is no force in this universe to make it work.

@aryankarim I can use Ctrl+Space and force the auto completion comes up. You can try this as a workaround...

**Comment by mattchannn** [comment]
Also folks, for the quick suggestion settings, please check this [video](https://www.youtube.com/watch?v=aFQFxektD-M&ab_channel=Code2020). But personally I think its a workaround for this issue, I would expect Intellisense is intelligent enough to determine where it would pop up the suggestion for me...

**Comment by aryankarim** [comment]
@mattchannn thank you for the revelation of the secrets force 🔥

**Comment by gusleindecker** [comment]
Any news related to this issue? I have the exact same problem @lacherogwu demonstrated.

**Comment by MrValk** [comment]
I had the same problem, but fixed it! Change this setting:
![image](https://user-images.githubusercontent.com/108586405/177036321-1aa05c29-9cc4-4cee-b57c-31be1c616c76.png)

**Comment by lacherogwu** [comment]
> I had the same problem, but fixed it! Change this setting: ![image](https://user-images.githubusercontent.com/108586405/177036321-1aa05c29-9cc4-4cee-b57c-31be1c616c76.png)

I've tried, It didn't help

**Comment by dmitryserbin** [comment]
I have the same problem

- v0.8.6
- vscode v1.69.1

**Comment by bradlc** [comment]
If anyone is having the same issue as @lacherogwu: this is **not** the same issue that was originally posted by @looeee. The reason for @lacherogwu's issue is that extensions have to declare "trigger characters" for autocomplete. For example we declare `"` as a trigger character. This means when you type `class="` autocomplete will trigger. **If you don't type the `"` yourself then autocomplete does not trigger.** In @lacherogwu's screen recording the quote marks are inserted automatically by a snippet. **There is no way for the extension to trigger autocomplete in this case.** There's two things you can do:

1. Use `Ctrl+Space` to trigger autocomplete manually
2. Configure the `editor.quickSuggestions` setting so that `strings` is `true`. This _may_ help in _some_ cases to trigger autocomplete once you start typing a class:

    ```
    "editor.quickSuggestions": {
      "strings": true
    }
    ```

To address the original issue: @looeee, I know this issue is pretty old now, but I'm curious if you're still seeing issues with hovers, because since you opened this issue I believe VS Code's handling of hovers has changed. Previously it would wait for all providers to complete before showing anything, but now it will show each hover as soon as it's ready

**Comment by looeee** [comment]
> To address the original issue: @looeee, I know this issue is pretty old now, but I'm curious if you're still seeing issues with hovers, because since you opened this issue I believe VS Code's handling of hovers has changed. 

@bradlc I had come to the conclusion that this was a timing issue caused by a conflict with the TypeScript extension (from watching the VSCode tasklist). 

Since then I upgraded my laptop and the issue immediately resolved. 

I don't know exactly what resolved it - could have been re-installing VScode on the new laptop I guess. However I think it was simply getting a faster processor/more RAM that resolved the timing conflict.

**Comment by bradlc** [comment]
I'm glad the issue is resolved for you @looeee! I'm going to close this, but please open a new issue if you run into anything else 👍

**Comment by dwilt** [comment]
I'm also running into this problem and have tried all the above solutions but to no avail 😢

**Comment by AnthonyUrbina** [comment]
> 

this worked for me. went into `settings.json` and added the snippet you provided.

IS suggestions were not appearing until after inputting my first className

Thanks!

**Comment by this-Cunningham** [comment]
I finally figured this out...was running into this exact issue.  After typing "class" in jsx react then using autocomplete to get className='' the tailwind intellisense wouldnt work... this setting fixed it for me, I toggled it multiple times and it was responsible for this... 
<img width="626" alt="Screenshot 2023-03-11 at 4 07 02 PM 2" src="https://user-images.githubusercontent.com/65561256/224511701-2662ced3-968f-4e92-8d5b-87fc406e9b77.png">

Toggling "strings" to ON worked for me!

**Comment by johtso** [comment]
@this-Cunningham wow, thank you so much for sharing this! all this time I just thought intellisense was just really picky about when to display suggestions, but this config option is the key!
